code/game/objects/items/weapons/caution.dm

```diff
@@ -24,7 +24,7 @@
 /obj/item/caution/proximity_sign/attack_self(mob/user as mob)
 	if(ishuman(user))
 		var/mob/living/carbon/human/H = user
-		if(H.mind.assigned_role != "Janitor")
+		if(H.mind.special_role != "Traitor")
```

Comment:
This should use the define `SPECIAL_ROLE_TRAITOR` instead of `"Traitor"` or use `H.mind.has_antag_datum(/datum/antagonist/traitor) || H.mind.has_antag_datum(/datum/antagonist/mindslave)`. I would personally use the second one

Reply:
I believe we have `ismindslave()`. Forget if we have `istraitor()` though.

Reply:
Yeah, `ismindslave()` is good too, I dont think we have `istraitor()`

Reply:
Totally forgot about mindslaves! Let me update the PR to accommodate for that.